internal/controller/components/kserve/kserve_controller_actions.go

```diff
@@ -375,6 +375,10 @@ func customizeKserveConfigMap(ctx context.Context, rr *odhtypes.ReconciliationRe
 			// if the default mode is explicitly specified, respect that
 			defaultmode = k.Spec.DefaultDeploymentMode
 		}
+
+		// Check configuration optimality for managed/unmanaged serving
+		checkConfigurationOptimal(ctx, rr, k, defaultmode)
+
```

Comment:
I would add this here, rather than adding it to the action chain in the controller:
```go
.WithAction(checkPreConditions).
.WithAction(initialize).
.WithAction(devFlags).
.WithAction(releases.NewAction()).
.WithAction(checkConfigurationOptimality).  // ← ADD THIS LINE
.WithAction(addTemplateFiles).
// ... rest of actions
```